### PR TITLE
ran out of time...

### PR DESCRIPTION
**Pull request by MikeyLHolm**
I ran out of time and ran into problem of not being able to change strings as they aren't mutable. I guess changing them to ASCII or initially read them as something else than STR would be more optimal. 

It's lacking still character check and the actually fall of the sand isn't finished. But it was meant to be double loop replacing " " with "." and "." with " " on Y + 1.

### CODE REVIEW

**Comment by mdesta** [comment]
I don't know if it is optimal, but I run into the same issue with strings being immutable, so my work around was to change it to a list and work on it, and finally join back list elements into a string.

File: chall04/mlindhol.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+# python 3.7.4 Windows 10
+
+import sys
+
+
+def input_error():
+    print("Bad input, n > 0 and only '.' ' ' and '#' allowed")
+    exit(-1)
```
Comment (jgengo) [inline chall04/mlindhol.py:9]: I would suggest avoiding creating a function for that and use sys.exit() with your message as argument of sys.exit(); 
https://docs.python.org/2/library/sys.html#sys.exit

File: chall04/mlindhol.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def main():
+    lines = []
+    lines = sys.stdin.read().splitlines()
```
Comment (jgengo) [inline chall04/mlindhol.py:14]:
```python
lines = sys.stdin.readlines()
```

File: chall04/mlindhol.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    n = int(lines[0])
+    if n < 1:
+        input_error()
+    lines.pop(0)
+
+    print("\n:::START POSITION:::\n")
+    print("N =", n)
```
Comment (jgengo) [inline chall04/mlindhol.py:21]: ?

File: chall04/mlindhol.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    print("--------------------")
+
+    for i in range(n):
+        for j in range(n):
+            if lines[i][j] is '.' and lines[i + 1][j] is ' ':
+                lines[i][j] = ' '
+                lines[i + 1][j] = '.'
+
+
+if __name__ == "__main__":
```
Comment (jgengo) [inline chall04/mlindhol.py:33]: good you protected it.

**Review by jgengo — COMMENTED** [review]
Good job!

**Pull request merged**